auth_service/auth_adapter/core/auth.py

```diff
@@ -55,32 +60,119 @@ class UserDataMismatchError(AuthAdapterError):
     """Raised when user claims do not match the registered user data."""
 
 
+class UserInfoError(AuthAdapterError):
+    """Error when retrieving data from the userinfo endpoint."""
+
+
+class ConfigurationDiscoveryError(ConfigurationMissingKey):
+    """Raised when configuration is missing and cannot be discovery."""
+
+
+class OIDCDiscovery:
+    """Helper class for using the OIDC discovery mechanism.
+
+    Configuration is only checked and data fetched over the network if needed,
+    and all network requests are cached.
+    """
+
+    _authority_url: Optional[str]
+
+    def __init__(self, authority_url: str):
+        """Initialize with the authority URL."""
+        if not authority_url or not authority_url.startswith("https://"):
+            self._authority_url = None
```

Comment:
Maybe raise an exception here?

Reply:
It is now required to be a URL in the config, so this check is not necessary.